Npm packages

RobinNagpal opened the pull request.

mviswanathsai committed:
Added content

Up for review

mviswanathsai committed:
Updated

mviswanathsai committed:
Update aave-npm-packages.yaml

RobinNagpal commented on src/aave-npm-packages.yaml:
```diff
@@ -7,35 +7,415 @@ created: "2022-08-02T04:15:13.911Z"
   - developer
   - engineering
-thumbnail: https://d31h13bdjwgzxs.cloudfront.net/guides/eth/aave/AAVE_NPM_Packages.png
+thumbnail: https://d31h13bdjwgzxs.cloudfront.net/guides/eth/aave/Borrowing_in_aave.png
```
No need to change the image

RobinNagpal commented on src/aave-npm-packages.yaml:
```diff
@@ -7,35 +7,415 @@ created: "2022-08-02T04:15:13.911Z"
+     You can run a full test suite using Docker Compose commands.
+
+     ## Aave coreV2
+     This is the previous version of Aave core V3. and so it has features similar to V3.
```
```suggestion
     package contains smart contracts for V2.
```

RobinNagpal commented on src/aave-npm-packages.yaml:
```diff
@@ -7,35 +7,415 @@ created: "2022-08-02T04:15:13.911Z"
+     This package contains the smart contracts source code and markets configuration related to Aave Protocol V3. 
+     It uses Docker Compose and Hardhat as development environment for compilation, testing and deployment tasks.
+
+     You can install @aave/aave-v3-core as an npm package to your hardhat or truffle workspace to get access to contracts and 
+     interfaces provided by Aave V3
+
+     The repository uses Docker Compose to manage sensitive keys and load the configuration. 
+     Before any action like test or deploy, you must run docker-compose up to start the contracts-env container, and then connect to the container console through docker-compose exec contracts-env bash.
+     You can run a full test suite using Docker Compose commands.
```
````suggestion

You can install `@aave/core-v3` as an NPM package in your Hardhat or Truffle project to import the contracts and interfaces:

`npm install @aave/core-v3`

Import at Solidity files:

```
import {IPool} from "@aave/core-v3/contracts/interfaces/IPool.sol";

contract Misc {

  function supply(address pool, address token, address user, uint256 amount) public {
    IPool(pool).supply(token, amount, user, 0);
    {...}
  }
}
```

The JSON artifacts with the ABI and Bytecode are also included into the bundled NPM package at `artifacts/` directory.

Import JSON file via Node JS `require`:

```
const PoolV3Artifact = require('@aave/core-v3/artifacts/contracts/protocol/pool/Pool.sol/Pool.json');

// Log the ABI into console
console.log(PoolV3Artifact.abi)
```
````

RobinNagpal commented on src/aave-npm-packages.yaml:
```diff
@@ -7,35 +7,415 @@ created: "2022-08-02T04:15:13.911Z"
-      List of packages
-      See here - https://www.npmjs.com/search?q=%40aave
+
+     ## Aave coreV3
```
```suggestion
     ## [@aave/core-v3](https://www.npmjs.com/package/@aave/core-v3)
```

RobinNagpal commented on src/aave-npm-packages.yaml:
```diff
@@ -7,35 +7,415 @@ created: "2022-08-02T04:15:13.911Z"
+     Before any action like test or deploy, you must run docker-compose up to start the contracts-env container, and then connect to the container console through docker-compose exec contracts-env bash.
+     You can run a full test suite using Docker Compose commands.
+
+     ## Aave coreV2
```
```suggestion
     ## [@aave/protocol-v2](https://www.npmjs.com/package/@aave/protocol-v2)
```

RobinNagpal commented on src/aave-npm-packages.yaml:
```diff
@@ -7,35 +7,415 @@ created: "2022-08-02T04:15:13.911Z"
+
+     ## Aave coreV2
+     This is the previous version of Aave core V3. and so it has features similar to V3.
+     The repository uses Docker Compose to manage sensitive keys and load the configuration. 
+     Prior any action like test or deploy, you must run docker-compose up to start the contracts-env container, and then connect to the container console via docker-compose exec contracts-env bash.
```
```suggestion
     Just like @aave/core-v3, you can  install @aave/protocol-v2 as an NPM package by running `npm install @aave/protocol-v2`
     
     You can import Solidity and ABI files, just like the V3 files shown above. 
```

RobinNagpal commented on src/aave-npm-packages.yaml:
```diff
@@ -7,35 +7,415 @@ created: "2022-08-02T04:15:13.911Z"
+     The repository uses Docker Compose to manage sensitive keys and load the configuration. 
+     Prior any action like test or deploy, you must run docker-compose up to start the contracts-env container, and then connect to the container console via docker-compose exec contracts-env bash.
+
+     ## Aave Stake
```
Use the correct npm package name and add couple of paragraphs for this.

https://www.npmjs.com/package/@aave/aave-stake

In the first paragraph explain about Safety module and in the second, high level about this repository

RobinNagpal commented on src/aave-npm-packages.yaml:
```diff
@@ -7,35 +7,415 @@ created: "2022-08-02T04:15:13.911Z"
     content: |
-      https://www.npmjs.com/package/@aave/periphery-v3
+
+     ## Aave Lens Protocol
```
We can skip lens protocol and move https://www.npmjs.com/package/@aave/aave-stake to this step

mviswanathsai committed:
Added changes according to yesterday's discussion

Not sure what to add in @aave/periphery-v3 , it has no readme file
Not sure whether or not to include @aave/aave-token

mviswanathsai committed:
Update aave-npm-packages.yaml

mviswanathsai committed:
Update aave-npm-packages.yaml

RobinNagpal committed:
Merge branch 'main' into npm-packages

# Conflicts:
#	src/aave-npm-packages.yaml

RobinNagpal committed:
run yarn test

RobinNagpal committed:
improve step1

RobinNagpal committed:
improve step1

RobinNagpal commented on src/aave-npm-packages.yaml:
```diff
@@ -11,30 +11,363 @@ publishStatus: Live
-      https://www.npmjs.com/package/@aave/math-utils
+     
+     ## @aave/protocol-js
+     AAVE is a decentralized non-custodial liquidity market protocol where users can participate as depositors or borrowers. 
```
Do you feel this line `AAVE is a decentralized non-custodial liquidity market protocol where users can participate as depositors or borrowers. ` add value to the user here? 

is this related to `@aave/protocol-js`? Why specifically here?

mviswanathsai replied:
My thought process was that since protocol-js encompasses the Aave protocol as a whole, it might be of value to add what the set of smart contracts provide the users with.

RobinNagpal commented on src/aave-npm-packages.yaml:
```diff
@@ -11,30 +11,363 @@ publishStatus: Live
+     The AAVE Protocol is a set of open source smart contracts which facilitate the lending and borrowing of user funds. These contracts, 
+     and all user transactions/balances are stored on a public ledger called a blockchain, making them accessible to anyone.
+
+     ethers js, something we discussed about earlier, is a peer dependancy (PeerDependency, it is not automatically installed. Instead, the code that includes the package must include it as its dependency.) for protocol-js.
```
How does this line connect to the previous line?

`ethers js, something we discussed about earlier, is a peer dependancy`

mviswanathsai replied:
discussed earlier in math-utils

     It formats two types of data, raw and indexed contract data, for front-end usage.
      * ethers.js: 
        ethers.js is considered raw contract data. ethers.js is a library for interacting with Ethereum and other EVM compatible blockchains.

RobinNagpal commented on src/aave-npm-packages.yaml:
```diff
@@ -11,30 +11,363 @@ publishStatus: Live
+     It also encloses various transaction methods and the **Lending pool V2**, of which we have talked about in detail. For a short summary,
+     Lending pool is just a smart contract which acts as a reserve (pool) of assets, lent by liquidity providers for the protocol to use
+     for loaning. It contains methods like deposit, borrow, repay, withdraw, etc which can be called by users via smartcontracts.
+     It also encloses other vast concepts like aave governance.
```
We wrote two paragraphs under `@aave/protocol-js` but nothing actually is related to `@aave/protocol-js` ?

RobinNagpal commented on src/aave-npm-packages.yaml:
```diff
@@ -11,30 +11,363 @@ publishStatus: Live
+     that can be freely moved within the underlying network. The holder of the tokenized position can redeem their share from the SM at any time, triggering a cooldown period of 
+     one week (which can be further extended by the governance, mentioned in the protocol-js).
+
+     ## @aave/protocol-v2
```
How is this different from @aave/coreV2 discussed in step1 ?

RobinNagpal commented on src/aave-npm-packages.yaml:
```diff
@@ -11,30 +11,363 @@ publishStatus: Live
+    stepItems:
+      - uuid: 881f3900-a654-4868-8d62-b80153494bf8
+        content: | 
+         @aave/protocol-js encloses which of these?
```
What does `encloses` mean?

RobinNagpal commented on src/aave-npm-packages.yaml:
```diff
@@ -11,30 +11,363 @@ publishStatus: Live
+          - C
+        type: MultipleChoice
+        choices:
+          - content: aave governance
```
```suggestion
          - content: Aave governance
```

RobinNagpal commented on src/aave-npm-packages.yaml:
```diff
@@ -11,30 +11,363 @@ publishStatus: Live
+        choices:
+          - content: aave governance
+            key: A
+          - content: aave lending pool
```
```suggestion
          - content: Aave lending pool
```
correct casing for the other options also

RobinNagpal commented on src/aave-npm-packages.yaml:
```diff
@@ -11,30 +11,363 @@ publishStatus: Live
+      * ethers.js: 
+        ethers.js is considered raw contract data. ethers.js is a library for interacting with Ethereum and other EVM compatible blockchains.
+
+      * subgraph:
```
How is this `subgraph` point related to @aave/math-utils?

mviswanathsai replied:
subgraphs and caching servers are used to collect input data for the math utils package to format

RobinNagpal commented on src/aave-npm-packages.yaml:
```diff
@@ -11,30 +11,363 @@ publishStatus: Live
+      
+      * Caching servers:
+        The Aave UI uses decentralized hosting with IPFS, which means that it doesn't require any API keys to run. However, this also means that only public rpc endpoints can be used, which are quickly rate-limited. 
+        To respond for this, Aave has published a caching server that performs contract queries on one server and then serves the data to all frontend users through a graphQL endpoint.
```
How is `Caching servers` related to `@aave/math-utils` ?

mviswanathsai committed:
Added changes 

updated math-utils step
updated the headings to include urls
updated wordings and capitalization

RobinNagpal committed:
Add aave risk management guides

RobinNagpal committed:
Merge remote-tracking branch 'origin/npm-packages' into npm-packages

RobinNagpal committed:
Merge latest code

mviswanathsai committed:
Added changes

removed aave tokens
added new content and questions for the other steps

mviswanathsai committed:
Update aave-npm-packages.yaml

RobinNagpal committed:
Merge branch 'main' into npm-packages

# Conflicts:
#	generated/json/aave-risk-management.json
#	generated/markdown/aave-risk-management.md
#	src/aave-risk-management.yaml

RobinNagpal committed:
Merge branch 'main' into npm-packages

# Conflicts:
#	generated/json/aave-risk-management.json
#	generated/markdown/aave-risk-management.md
#	src/aave-risk-management.yaml

RobinNagpal committed:
Update guide

RobinNagpal commented on src/aave-npm-packages.yaml:
```diff
@@ -11,30 +11,591 @@ publishStatus: Live
+             key: C
+           - content: delegate
+             key: D
+      - uuid: 9a8229eb-f548-4596-b938-7d7f63dc6adb
```
Is 'C' the right option?  setUsageAsCollateral see to set something rather then determine

RobinNagpal replied:
Please reword the options for better clarity

RobinNagpal commented on src/aave-npm-packages.yaml:
```diff
@@ -11,30 +11,591 @@ publishStatus: Live
+            key: A
+          - content: Use the collateral to repay loan
+            key: B
+          - content: Withdraw all of the collateral on command
```
Reword this option `Withdraw all of the collateral on command` as its not clear

RobinNagpal commented on src/aave-npm-packages.yaml:
```diff
@@ -11,30 +11,591 @@ publishStatus: Live
+            key: D
+
+      - uuid: 4aa31aea-1718-433c-9ebf-44cf22b442c7
+        content: Methods that concern with collaterals, allow the user to do which of the following?
```
```suggestion
        content: What operations related to collaterals can the user perform using `@aave/protocol-js`?
```

RobinNagpal committed:
Some cleanups
